File: lib/scrapers/scraper_registry.dart
import '../services/logger_service.dart';
import 'base_scraper.dart';
import 'dynamic_webview_scraper.dart';
import 'reddit_scraper.dart';
import 'static_board_scraper.dart';

class ScraperRegistry {
  ScraperRegistry({LoggerService? logger})
    : scrapers = [
        RedditScraper(logger: logger),
        StaticBoardScraper(logger: logger),
        DynamicWebViewScraper(logger: logger),
      ];

  final List<BaseScraper> scrapers;

  BaseScraper forUrl(String url) => selectScraper(url, scrapers);
}

File: app/src/main/java/io/github/rustnomicon/sourceinstaller/scrape/ScraperRegistry.kt
package io.github.rustnomicon.sourceinstaller.scrape

import io.github.rustnomicon.sourceinstaller.service.LoggerService

class ScraperRegistry(
    logger: LoggerService = LoggerService.getInstance(),
    val scrapers: List<BaseScraper> = listOf(
        RedditScraper(logger = logger),
        StaticBoardScraper(logger = logger),
        DynamicWebViewScraper(logger = logger),
    ),
) {
    fun forUrl(url: String): BaseScraper = selectScraper(url, scrapers)
}
